src/github_domain.py

```diff
@@ -0,0 +1,138 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects for github entities."""
+
+from __future__ import annotations
+
+import datetime
+
+from typing import Any, Dict, List, Optional, Type
+
+DEFAULT_TIMESTAMP: datetime.datetime = (
+    datetime.datetime(datetime.MINYEAR, 1, 1, tzinfo=datetime.timezone.utc))
+
+
+class Assignee:
+    """A class representing an assignee of a pull request."""
+
+    def __init__(
+        self,
+        username: str,
+        timestamp: datetime.datetime = DEFAULT_TIMESTAMP
+    ) -> None:
+        """
+        Args:
+            username: str. GitHub username of the assignee.
+            timestamp: datetime. The time when an event happened.
+        """
+        self.username = username
+        self.timestamp = timestamp
+
+    def set_timestamp(self, timestamp: datetime.datetime) -> None:
+        """Sets timestamp to assignee.
+        Args:
+            timestamp: datetime. the time when an event happened.
+        """
+        self.timestamp = timestamp
+
+    def get_readable_waiting_time(self) -> str:
+        """Returns readable waiting time on review.
+        waiting time = ( current time - when reviewer was assigned )
+
+        Returns:
+            str. the waiting time in days and hours.
+        """
+        delta = datetime.datetime.now(datetime.timezone.utc) - self.timestamp
+        days = delta.days
+        hours, _ = divmod(delta.seconds, 3600)
+        waiting_time = []
+        if days:
+            waiting_time.append(f'{days} day{"s" if days > 1 else ""}')
+
+        if hours:
+            waiting_time.append(f'{hours} hour{"s" if hours > 1 else ""}')
+
+        return ', '.join(waiting_time)
+
+    def __repr__(self) -> str:
+        return f'@{self.username} assigned on {self.timestamp}'
+
+
+class PullRequest:
+    """A class representing a pull request on github."""
+
+    def __init__(
+        self,
+        url: str,
+        number: int,
+        author_username: str,
+        title: str,
+        assignees: List[Assignee]
+    ) -> None:
+        """Constructs a PullRequest object.
+
+        Args:
+            url: str.
+            number: int. The PR number.
+            author_username: str. github username of the Author.
+            title: str. PR title.
+            assignees: List(Assignee). List of assignees.
+        """
+        self.url = url
+        self.number = number
+        self.author_username = author_username
+        self.title = title
+        self.assignees = assignees
+
+    def is_reviewer_assigned(self) -> bool:
+        """Checks whether a reviewer is assigned to the PR.
+        Returns:
```

Comment:
Ditto, add one newline above. Please fix all other occurrences in this PR.

Reply:
done. This time I ran the lint checks from `oppia/oppia` repo on the code in this PR.